client/core/bond.go

```diff
@@ -0,0 +1,732 @@
+	// retry like this...
+	schedRetry := func() {
+		time.AfterFunc(20*time.Second, func() {
+			c.postAndConfirmBond(dc, bond)
+		}) // not ideal - supervisory goroutine TODO
+		c.log.Warnf("Server already confirming bond %v (%s). Will retry.",
+			coinIDStr, unbip(assetID))
+	}
+
+	// Inform the server, which will attempt to locate the bond and check
+	// confirmations. If server sees the required number of confirmations, the
+	// bond will be active (and account created if new) and we should confirm
+	// the bond (in DB and dc.acct.{bond,pendingBonds}).
+	pbr, err := c.postBond(dc, bond) // can be long while server searches
+	if err != nil {
+		var mErr *msgjson.Error
+		if errors.As(err, &mErr) && mErr.Code == msgjson.BondAlreadyConfirmingError {
+			// The server is already handling a postbond request for this bond.
+			// If we still have a response handler for the initial request, we
+			// will get the response. If not (restart or previous request
+			// timeout), we can either reconnect or postbond again *after*
+			// server has confirmed it.
+			schedRetry()
+			return
+		}
+		details := fmt.Sprintf("postbond request error: %v", err)
+		c.notify(newBondPostNote(TopicBondPostError, string(TopicBondPostError),
+			details, db.ErrorLevel, dc.acct.host))
+		return
+	}
+
+	if pbr.Active {
+		c.log.Infof("Bond confirmed %v (%s) with expire time of %v", coinIDStr,
+			unbip(assetID), time.Unix(pbr.Expiry, 0))
+		err = c.bondConfirmed(dc, assetID, coinID, pbr.Tier)
+		if err != nil {
+			c.log.Errorf("Unable to confirm bond: %v", err)
+		}
+		return
+	}
+
+	c.log.Warnf("Server still doesn't see the bond as fully confirmed: %s (%s)!",
+		coinIDStr, unbip(assetID))
+	schedRetry()
+}
+
+// monitorBondConfs launches a block waiter for the bond txns to reach the
+// required amount of confirmations. Once the requirement is met the server is
+// notified.
+func (c *Core) monitorBondConfs(dc *dexConnection, bond *asset.Bond, reqConfs uint32) {
+	assetID, coinID := bond.AssetID, bond.CoinID
+	coinIDStr := coinIDString(assetID, coinID)
+	host := dc.acct.host
+
+	wallet, _ := c.connectedWallet(assetID)
+	if wallet == nil {
+		c.log.Errorf("No wallet for asset %v", unbip(assetID))
+		return
+	}
+	lastConfs, err := wallet.RegFeeConfirmations(c.ctx, coinID)
+	coinNotFound := errors.Is(err, asset.CoinNotFoundError)
+	if err != nil && !coinNotFound {
+		c.log.Errorf("Error getting confirmations for %s: %w", coinIDStr, err)
+		return
+	}
+
+	if lastConfs >= reqConfs { // don't bother waiting for a block
+		go c.postAndConfirmBond(dc, bond)
+		return
+	}
+
+	if coinNotFound {
+		// Broadcast the bond and start waiting for confs.
+		c.log.Infof("Broadcasting bond %v (%s), script = %x.\n\n"+
+			"BACKUP refund tx paying to current wallet: %x\n\n",
+			coinIDStr, unbip(bond.AssetID), bond.BondData, bond.RedeemTx)
+		if _, err = wallet.SendTransaction(bond.SignedTx); err != nil {
+			c.log.Warnf("Failed to broadcast bond txn: %v")
```

Comment:
Feels funny to use this as the indicator for whether to broadcast the bond. Lemme see if I've got the reasoning. If 1) this is the first time we're broadcasting, this should obviously work. If 2) we are picking up monitoring on a transaction that had already been broadcast, **but has not been picked up by the server's node**, then a) If we're using a full node, we should get an error, which will be logged as just a warning. If b) we're using an SPV node, there will be no error or log. In the end, the result is the same, worst that can happen is a warning. Sound right?

Reply:
Yah the reasoning is right, but fair point that this feels funny to start this off with with a CoinNotFound -> YOLO broadcast.  Since the broadcast should have been done in `(c *Core) PostBond` first, the actions here *shouldn't* be necessary except for weird edges like clearing full node mempool, spv wallet recovery (wiping data files), etc.
I suspect it would be alright without this though.  Will revisit to see if I had other reasoning that I failed to doc.